src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+
+    /**
+     * Pointer to first node.
+     */
+    private MyLinkedList.Node<T> first;
+
+    /**
+     * Pointer to last node.
+     */
+    private MyLinkedList.Node<T> last;
+
+    public MyLinkedList() {
+    }
+
+    T unlink(MyLinkedList.Node<T> x) {
+        // assert x != null;
+        final T element = x.item;
+        final MyLinkedList.Node<T> next = x.next;
+        final MyLinkedList.Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            x.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        x.item = null;
+        size--;
+        return element;
+    }
+
     @Override
     public void add(T value) {
+        final MyLinkedList.Node<T> l = last;
+        final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(l, value, null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. Also, the `Node` class should be an inner class, so the explicit reference to `MyLinkedList.Node` is unnecessary.